tests/test_schema_dict_fields_generator.py

```diff
@@ -0,0 +1,33 @@
+import json
+
+import pytest
+
+from libcove2.common import schema_dict_fields_generator
+
+
+@pytest.fixture
+def schema_0_3():
+    """Schema for BODS 0.3"""
+    with open("tests/fixtures/schema-0-3-0.json", "r") as read_file:
+        return json.load(read_file)
+
+
+@pytest.fixture
+def schema_0_4():
+    """Schema for BODS 0.4"""
+    with open("tests/fixtures/schema-0-4-0.json", "r") as read_file:
+        return json.load(read_file)
+
+
+def test_schema_dict_fields_generator_0_3(schema_0_3):
+
+    schema_fields = set(schema_dict_fields_generator(schema_0_3))
+
+    assert len(schema_fields) == 130
```

Comment:
There must be a bug in the code. There should be 138 fields. These aren't returned:

```python
{'/annotations/url', '/annotations/createdBy', '/annotations/statementPointerTarget', '/annotations/createdBy/uri', '/annotations/creationDate', '/annotations/description', '/annotations/createdBy/name', '/annotations/transformedContent'}
```

Reply:
Ah, the bug is that `annotations/items` sets both `properties` and `oneOf`, but `_process_items` treats these as alternatives (`elif`).

Reply:
@jpmckinney Thanks for spotting this out. There is a new PR to fix it: https://github.com/OpenDataServices/lib-cove-2/pull/14 and more tests have been added.